File: gearGauge/gearGauge/Models/GearType.swift
//
//  GearType.swift
//  gearGauge
//
//  Created by Shaun Hutchinson on 03/11/2025.
//

import SwiftUI

enum GearType: String, CaseIterable, Identifiable {
    case shoes
    case bicycle

    var id: String { self.rawValue }

    var displayName: String {
        switch self {
        case .shoes:
            return String(localized: .gearTypeShoes)
        case .bicycle:
            return String(localized: .gearTypeBicycle)
        }
    }

    var displayIcon: String {
        switch self {
        case .shoes:
            return "shoe"
        case .bicycle:
            return "bicycle"
        }
    }
}

File: gear-gauge-rn/src/models/GearType.ts
/**
 * GearType — categorises a piece of fitness gear.
 * Mirrors the Swift `GearType` enum in gearGauge/gearGauge/Models/GearType.swift
 */
export enum GearType {
  Shoes = 'shoes',
  Bicycle = 'bicycle',
}

/** Display metadata associated with each GearType variant. */
export interface GearTypeMeta {
  /** Human-readable label (localisable in the future). */
  displayName: string;
  /** SF Symbol / icon name used to represent this gear type in the UI. */
  displayIcon: string;
}

/**
 * Lookup map providing `displayName` and `displayIcon` for every GearType.
 * TS enums cannot hold methods, so metadata lives in this separate const.
 */
export const GEAR_TYPE_META: Record<GearType, GearTypeMeta> = {
  [GearType.Shoes]: {
    displayName: 'Shoes',
    displayIcon: 'shoe',
  },
  [GearType.Bicycle]: {
    displayName: 'Bicycle',
    displayIcon: 'bicycle',
  },
};

/** Convenience accessor — returns display metadata for the given GearType. */
export function getGearTypeMeta(type: GearType): GearTypeMeta {
  return GEAR_TYPE_META[type];
}
